scalyr_agent/scalyr_client.py

```diff
@@ -1506,28 +1522,32 @@ def __set_attributes(self, thread_id, attributes):
         self.__thread_id = thread_id
         self.__attrs = attributes
         # A new event.  We have to create the serialization base using provided information/
-        # 2->TODO: use BytesIO, make all written data - binary.
-        tmp_buffer = StringIO()
+        # 2->TODO: should it be bytes, or it will be better to leave it  as unicode and just convert it
```

Comment:
In Python3 json libraries return only unicode, I am pretty sure. That's why I actually have done in the opposite way because I wanted to keep python2 as close as possible to python3 behavior.
https://github.com/scalyr/scalyr-agent-2/blob/76d9e328aea6e21b1f0699e5fdcfe5a89be13101/scalyr_agent/util.py#L145-L156